rwengine/src/engine/GameData.cpp

```diff
@@ -290,7 +290,7 @@ void GameData::loadGXT(const std::string &name)
 void GameData::loadWaterpro(const std::string& path)
 {
 	auto syspath = index.findFilePath(path);
-	std::ifstream ifstr(syspath.c_str(), std::ios_base::binary);
+	std::ifstream ifstr((const char*)syspath.c_str(), std::ios_base::binary);
```

Comment:
This should just use .string() instead of casting, c_str() will be returning wide characters here. The same applies elsewhere the code calls .c_str() on paths.